### ISSUE TITLE
asd

### ISSUE THREAD

**Issue by LucasHagen** [checkstyle/checkstyle#8767]
**Is your feature request related to a problem? Please describe.**

I'm trying to make the checkstyle JavaDoc verification match my team's coding style, but there is still one option that is not possible. I can't disable the rule that requires a new line before the `<p>` tag.

**Describe the solution you'd like**

I would like to request to allow the following paragraph style on JavaDocs:

```java
    /**
     * My description.
     * <p>
     * My class details.
     */
    public MyClass() {
        ...
    }
```

A simple suggestion for the config file:

```xml
        <module name="JavadocParagraph">
            <property name="allowNewlineParagraph" value="false"/>
            <property name="requireNewlineBefore" value="false"/> <!-- Requested option -->
        </module>
```

This new rule should make sure the `<p>` tag is in a new line, but should not require a new line before it.

**Additional context**

When I try to use the java doc using the following configuration, I get this error message:

```xml
        <module name="JavadocParagraph">
            <property name="allowNewlineParagraph" value="false"/>
        </module>
```

```bash
[ERROR] /my/path/to/my/class/MyClass.java:34: <p> tag should be preceded with an empty line. [JavadocParagraph]
```

Thanks!

**Comment by ramonsaboya** [comment]
+1

**Comment by romani** [comment]
https://checkstyle.org/checks/javadoc/javadocparagraph.html#JavadocParagraph is very focused on google style.

if you do not like it you should not use this Check.
if you do not like specific validation in this Check, you can suppress violation based on Check name and violation message.

look at https://checkstyle.org/filters/suppressionsinglefilter.html#SuppressionSingleFilter
add yo your config something like:
```
  <module name="SuppressionSingleFilter">
    <property name="checks" value="JavadocParagraph"/>
    <property name="message" value="tag should be preceded with an empty line"/>
  </module>
```

**Comment by cowwoc** [comment]
@romani I get it, but at the same time there is no rule (that I can see) for enforcing the JDK style which many of us prefer:

```
/**
 * My description.
 * <p>
 * My class details.
 */
```

Wouldn't it be easier to add an option to this rule than creating a whole new one?

**Comment by romani** [comment]
>enforcing the JDK style which many of us prefer:

Please share link to let me read it.

-------

This issue becomes confusing. We started from Google style config, that is designed to address what is declared in Google style. Google style is very unique in requirements in many cases, I am not sure I would recommend all use it blindly and strictly.

If someone try to mostly follow Google style, but they don't like some specific validation, they need to use suppressions or create their own config and adjust Check as they need.

And now we talk about Jdk style.
@cowwoc , if your concern is not about Jdk style, let's create new issue.

**Comment by romani** [comment]
@LucasHagen , @ramonsaboya , 
Does suppression work for you? 

If not I need more details on what you need to may decision on how to reconcile all users requirements
